build.gradle.kts

```diff
@@ -139,3 +139,20 @@ tasks.withType<KotlinCompile> {
 }
 
 
+
+tasks.register("resolveAndLockAll") {
+    doFirst {
+        require(gradle.startParameter.isWriteDependencyLocks)
+    }
+    doLast {
+        configurations.filter {
+            // Add any custom filtering on the configurations to be resolved
```

Comment:
Nice comment :P

Reply:
imagine writing such comments. :D This whole code is copy pasta from the gradle docs. :D